### PR TITLE
mocha in CI

### LINKED ISSUE #50: Testing: run mocha tests in ci
We should add the mocha tests to run in CI

### PR DESCRIPTION
**Pull request by 0xJepsen**
Closes #50

### CODE REVIEW

**Comment by 0xJepsen** [comment]
Edit: in order to do this we need to download and build the circom compiler from source in the CI workflow (not ideal)

File: circuits/test/gfmulint/gfmulint.test.ts [hunk 1]
```diff
@@ -21,31 +19,16 @@ describe("gfmulint", () => {
...
+    /// x^14 + x^4 + x^2 + 1 -> 10001010000010 is the right most 15 bits of the expected result
+    let expected_bit_array = [1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0];
 
-function parse_arr128_to_number(res: Arr128): number {
-  let first_64: number[] = res[0];
-  let last_64: number[] = res[1];
-  let all_bits: number[] = first_64.concat(last_64);
-  return parseInt(all_bits.join(""), 2);
-}
+    await circuit.expectPass( { a: a_bits, b: a_bits }, { res: expected_bit_array });
```
Comment (KaiGeffen) [inline circuits/test/gfmulint/gfmulint.test.ts:32]: I think that using `compute` over `expectPass` is better for interpretability of the results and is in line with expected behavior of testing frameworks that will say something like:
```
test failed
expect: foo
got: bar
```

File: circuits/test/common/index.ts [hunk 2]
```diff
@@ -30,6 +30,23 @@ export function hexToBitArray(hex: string): number[] {
   );
 }
 
+
+export function pad_num_to_arr128(value: number): number[][] {
```
Comment (KaiGeffen) [inline circuits/test/common/index.ts:34]: Pulling out these utility methods looks good.

**Review by KaiGeffen — CHANGES_REQUESTED** [review]

File: circuits/test/gfmulint/gfmulint.test.ts (hunk 1 above)
Comment (0xJepsen) [reply circuits/test/gfmulint/gfmulint.test.ts:32]: i will probably close the pr for now since tackling the issue isn't ideal anymore

Comment (thor314) [reply circuits/test/gfmulint/gfmulint.test.ts:32]: oh why not? I likeed the CI

**Pull request closed without merging**